src/mrsimulator/method/plot.py

```diff
@@ -0,0 +1,343 @@
+COLORS = list(mcolors.TABLEAU_COLORS)
+EVENT_COLORS = {  # TODO: add colors
+    "ConstantDurationEvent": "orange",
+    "SpectralEvent": "g",
+    "MixingEvent": "b",
+}
+GRAPH_PROPERTIES = {
+    "rotor_angle": {
+        "ylim": (0, 90),
+        "yticks": [15, 30, 45, 60, 75],
+        "grid": {"axis": "y", "color": "black", "alpha": 0.2},
+    }
+}
+
+
+def _make_x_data(df):
+    """Returns list of x points to use in plotting"""
+    points = [0]
+
+    for i, row in df.iterrows():
+        if row["type"] == "SpectralEvent":
+            next_x = points[-1] + (row["fraction"] * SPECTRAL_MULTIPLIER)
+            points.extend((next_x, next_x))
+        elif row["type"] == "ConstantDurationEvent":
+            next_x = points[-1] + DURATION_WIDTH
+            points.extend((next_x, next_x))
+
+    points.pop()
+    return points
+
+
+def _offset_x_data(df, x_data):
+    """Offsets x_data based on MixingEvents"""
+    offset_x = np.array([0] + x_data)
+    ev_groups = [(_type, sum(1 for _ in group)) for _type, group in groupby(df["type"])]
+
+    # Remove MixingEvents from end of sequence
+    if ev_groups[-1][0] == "MixingEvent":
+        ev_groups.pop()
+
+    df_idx = 0
+    # Extend first jump if first event(s) are MixingEvent
+    if ev_groups[0][0] == "MixingEvent":
+        gp__ = ev_groups[0][1]
+        offset = sum(df["tip_angle"][0:gp__]) / 360.0 * MIXING_WIDTH
+        offset_x[1] += offset
+        # Increment event indexer by number of MixingEvents in first group
+        df_idx += gp__
+        ev_groups.pop(0)
+
+    x_idx = 1
+    for _type, num in ev_groups:
+        if _type == "MixingEvent":
+            up_lim__ = df_idx + num
+            offset = sum(df["tip_angle"][df_idx:up_lim__]) / 360.0 * MIXING_WIDTH
+            offset_x[x_idx] -= offset / 2
+            offset_x[x_idx + 1] += offset / 2
+            x_idx += 1
+        else:
+            if offset_x[x_idx] == offset_x[x_idx + 1]:
+                x_idx += 1
+            x_idx += (num * 2) - 1
+        # Increment event indexer by number of Events in group
+        df_idx += num
+
+    return offset_x
+
+
+def _add_rect_with_label(ax, x0, x1, label, ev_type):
+    """Add a rectangle between x0 and x1 on ax representing event"""
+    rect_kwargs = {"color": EVENT_COLORS[ev_type], "alpha": 0.2}
+    anno_kwargs = {
+        "color": "black",
+        "ha": "center",
+        "va": "center",
+        "fontsize": 8,
+        "rotation": 90 if ev_type == "MixingEvent" else 0,
+    }
+    rect = Rectangle((x0, 0), x1 - x0, 1, **rect_kwargs)
+    ax.add_patch(rect)
+    if label is not None:
+        ax.annotate(label, ((x1 + x0 + 0.015) / 2, 0.5), **anno_kwargs)
+
+
+def _format_mix_label(df, df_idx, j):
+    """Helper method to format label for mixing events."""
+    tip_angle = df["tip_angle"][df_idx + j]
+    phase = df["phase"][df_idx + j]
+    return "({0:.1f}, {1:.1f})".format(tip_angle, phase), tip_angle / 360 * MIXING_WIDTH
```

Comment:
The arguments don't need to be df, df_idx, and j, when all you need is the tip_angle and phase. The function will be a lot simpler then.